### DISCUSSION TITLE
Redirects for non-existant short urls goes to full YOURLS directory

### DISCUSSION THREAD

**Discussion by adamesq** [YOURLS/YOURLS#3584] [Q&A] [answered]
I moved my YOURLS earlier today from one domain to another on the same account. The only issue I'm having is when I put in a shortened URL that I haven't created, I get redirected to the YOURLS main directory. All legit short URLs work as they should. Admin login works as it should. Any ideas? THANX!

**Comment by dgw** [comment, marked as answer]
What you describe is YOURLS' default behavior when a keyword isn't defined yet:

https://github.com/YOURLS/YOURLS/blob/f1ae0efd7ad26ec578c283d37d20ad67129f1887/yourls-go.php#L19-L27

…likely combined with a webserver that is configured to show a directory listing when there's no index file (`index.html`, `index.php`, etc.), based on your use of the phrase "full YOURLS directory".

Disabling directory indexes on your YOURLS domain is recommended. To prevent missing keywords from redirecting to the domain root, and make them directly return Not Found, install and activate [the 404-if-not-found plugin](https://github.com/YOURLS/404-if-not-found).

**Reply by adamesq** [reply]
THANK YOU again, @dgw! I read a reply you had written many years ago before posting. Somehow, this time in sunk in! 😂
